File: app/src/main/java/com/sqube/tipshub/PostFragment.java
package com.sqube.tipshub;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.clans.fab.FloatingActionButton;
import com.github.clans.fab.FloatingActionMenu;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;

import adapters.PostAdapter;

public class PostFragment extends Fragment {
    private FirebaseFirestore database;
    private Query query;
    private FirebaseAuth auth;
    private FirebaseUser user;
    private final String TAG = "PostFragment";
    String userId, myId, myUsername;
    PostAdapter postAdapter;
    FloatingActionButton fapTip, fabNormal;
    FloatingActionMenu fabMenu;
    RecyclerView recyclerView;
    Intent intent;

    public PostFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView  = inflater.inflate(R.layout.fragment_post, container, false);
        recyclerView = rootView.findViewById(R.id.postList);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        database = FirebaseFirestore.getInstance();
        auth = FirebaseAuth.getInstance();
        user = auth.getCurrentUser();
        myId = user.getUid();
        myUsername = user.getDisplayName();
        userId = getArguments().getString("userId");
        loadPost();
        return rootView;

    }
    private void loadPost() {
        Log.i(TAG, "loadPost: ");
        query = database.collection("posts").orderBy("time", Query.Direction.DESCENDING).whereEqualTo("userId", userId);
        postAdapter = new PostAdapter(query, myId, getActivity(), getContext());
        recyclerView.setAdapter(postAdapter);
        if(postAdapter!=null){
            Log.i(TAG, "loadPost: started listening");
            postAdapter.startListening();
        }
    }
}

File: app/src/main/java/com/sqube/tipshub/fragments/PostFragment.kt
package com.sqube.tipshub.fragments

import adapters.PostAdapter
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.Query
import com.sqube.tipshub.databinding.FragmentPostBinding
import com.sqube.tipshub.models.Post
import com.sqube.tipshub.utils.FirebaseUtil
import com.sqube.tipshub.utils.GUEST

class PostFragment : Fragment() {
    private var userId: String? = null
    private lateinit var myId: String
    private var user: FirebaseUser? = null
    private var postAdapter: PostAdapter? = null
    private var _binding: FragmentPostBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        _binding = FragmentPostBinding.inflate(inflater, container, false)
        (binding.postList.itemAnimator as DefaultItemAnimator?)!!.supportsChangeAnimations = false
        binding.postList.layoutManager = LinearLayoutManager(activity)
        user = FirebaseUtil.firebaseAuthentication?.currentUser
        myId = user?.uid ?: GUEST
        userId = arguments?.getString("userId")
        loadPost()
        return binding.root
    }

    override fun onResume() {
        super.onResume()
        user = FirebaseUtil.firebaseAuthentication?.currentUser
        myId = if (user == null) GUEST else user!!.uid
        if (postAdapter != null) postAdapter!!.setUserId(myId)
    }

    private fun loadPost() {
        val TAG = "PostFragment"
        Log.i(TAG, "loadPost: ")
        val query = FirebaseUtil.firebaseFirestore?.collection("posts")!!.orderBy("time", Query.Direction.DESCENDING).whereEqualTo("userId", userId)
        val response = FirestoreRecyclerOptions.Builder<Post>()
                .setQuery(query, Post::class.java)
                .build()
        postAdapter = PostAdapter(response, myId, requireContext())
        binding.postList.adapter = postAdapter
        if (postAdapter != null) {
            Log.i(TAG, "loadPost: started listening")
            postAdapter!!.startListening()
        }
    }
}